tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -276,6 +276,27 @@ def augmented_dickey_fuller(x):
 
 @set_property("fctype", "aggregate")
 @not_apply_to_raw_numbers
+def augmented_dickey_fuller_p_val(x):
+    """
+    The Augmented Dickey-Fuller is a hypothesis test which checks whether a unit root is present in a time
+    series sample. This feature calculator returns the value of the corresponding p-value.
+
+    :param x: the time series to calculate the feature of
+    :type x: pandas.Series
+    :return: the value of this feature
+    :return type: float
+    """
+    maxlag = 1
```

Comment:
Line 290 to 295 should be moved into its own method and wrapped with a cache decorator